nixos/modules/config/users-groups.nix

```
@@ -473,14 +475,15 @@ in {
       '';
     };
 
-    # FIXME: obsolete - will remove.
-    security.initialRootPassword = mkOption {
-      type = types.str;
-      default = "!";
-      example = "";
-      visible = false;
-    };
 
+    users.allowLoginless = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to allow being locked out of your system by not allowing
+        login to any administrator user.
```

This phrasing makes it sound strange that this option even exists (why would you ever want to allow being locked out of your system?).

Maybe something like:
```suggestion
        Do not require that at least the `root` user or a user in the `wheel` group can log in using 
        a password or an SSH key.
```

---

https://github.com/NixOS/nixpkgs/issues/73003 mentions several use cases:
* Allowing login only via ssh keys
* Hardening a container